podpac/core/coordinates/dependent_coordinates.py

```diff
@@ -0,0 +1,452 @@
+        if segment_lengths is not None:
+            self._set_segment_lengths(segment_lengths)
+        else:
+            self.segment_lengths # force validation
+
+    @tl.default('dims')
+    def _default_dims(self):
+        return tuple(None for c in self.coordinates)
+
+    @tl.default('idims')
+    def _default_idims(self):
+        return tuple('ijkl')[:self.ndims]
+
+    @tl.default('ctypes')
+    def _default_ctype(self):
+        return tuple('point' for dim in self.dims)
+
+    @tl.default('units')
+    def _default_units(self):
+        return tuple(None for dim in self.dims)
+
+    @tl.default('segment_lengths')
+    def _default_segment_lengths(self):
+        return tuple(None for dim in self.dims)
+
+    @tl.default('coord_ref_sys')
+    def _default_coord_ref_sys(self):
+        return settings['DEFAULT_COORD_REF_SYS']
+
+    @tl.validate('coordinates')
+    def _validate_coordinates(self, d):
+        val = d['value']
+        if len(val) == 0:
+            raise ValueError("Dependent coordinates cannot be empty")
+
+        for i, a in enumerate(val):
+            if a.shape != val[0].shape:
+                raise ValueError("coordinates shape mismatch at position %d, %s != %s" % (
+                    i, a.shape, val[0].shape))
+        return val
+
+    @tl.validate('dims')
+    def _validate_dims(self, d):
+        val = self._validate_sizes(d)
+        for i, dim in enumerate(val):
+            if dim is not None and dim in val[:i]:
+                raise ValueError("Duplicate dimension '%s' in stacked coords" % dim)
+        return val
+
+    @tl.validate('segment_lengths')
+    def _validate_segment_lengths(self, d):
+        val = self._validate_sizes(d)
+        for i, (segment_lengths, ctype) in enumerate(zip(val, self.ctypes)):
+            if segment_lengths is None:
+                if ctype != 'point':
+                    raise TypeError("segment_lengths cannot be None for '%s' coordinates at position %d" % (ctype, i))
+            else:
+                if ctype == 'point':
+                    raise TypeError("segment_lengths must be None for '%s' coordinates at position %d" % (ctype, i))
+                if segment_lengths <= 0.0:
+                    raise ValueError("segment_lengths must be positive at pos %d" % i)
+        return val
+
+    @tl.validate('idims', 'ctypes', 'units')
+    def _validate_sizes(self, d):
+        if len(d['value']) != self.ndims:
+            raise ValueError("%s and coordinates size mismatch, %d != %d" % (
+                d['trait'].name, len(d['value']), self.ndims))
+        return d['value']
+
+    @tl.observe('dims', 'idims', 'ctypes', 'units', 'coord_ref_sys', 'segment_lengths')
+    def _set_property(self, d):
+        self._properties.add(d['name'])
+
+    def _set_name(self, value):
+        # only set if the dims have not been set already
+        if 'dims' not in self._properties:
+            self.set_trait('dims', value.split(','))
```

Comment:
Can we support 'lat,lon' and 'lat, lon'?  i.e. after the split map each value to get rid of preceding whitespace? I know this is annoying, but its the first thing I did when I tried to use this.